shunt/vxlan_over_ssh.py

```diff
@@ -0,0 +1,145 @@
+    shellcmd.run_cmd("sudo ip link add %s type dummy" % (if_name))
+    shellcmd.run_cmd("sudo ip addr add %s/24 dev %s" % (if_ip, if_name))
+    shellcmd.run_cmd("sudo ip link set %s up" % (if_name))
+
+
+def socat_tcp_to_udp(src_port, dst_port):
+    """
+    Socat command to map TCP port to UDP port
+    Args:
+        src_port: Source TCP port
+        dst_port: Destination UDP port
+    Returns: None
+    """
+    shellcmd = ShellCommandHelper()
+    shellcmd.run_cmd("socat -T15 tcp4-listen:%s,reuseaddr,fork udp:localhost:%s"
+                     % (src_port, dst_port), detach=True)
+
+
+def socat_udp_to_tcp(src_port, dst_port):
+    """
+    Socat command to map UDP port to TCP port
+    Args:
+        src_port: Source UDP port
+        dst_port: Destincation TCP port
+    Returns: None
+    """
+    shellcmd = ShellCommandHelper()
+    shellcmd.run_cmd("socat -T15 udp4-listen:%s,reuseaddr,fork tcp:localhost:%s"
+                     % (src_port, dst_port), detach=True)
+
+
+def iptables_udp_change_source(dst_ip, dst_port, src_ip_port):
+    """
+    iptables command to change source IP/ports of UDP packets
+    Args:
+        dst_ip: Destination IP of packet to be changed
+        dst_port: Destination port of packet to be changed
+        src_ip_port: <source_ip_to_be_added>:<source port range> e.g. 192.168.1.2:38000-45000
+    Returns: None
+    """
+    shellcmd = ShellCommandHelper()
+    shellcmd.run_cmd(
+        "sudo iptables -t nat -A POSTROUTING -p udp -d %s --dport %s -j SNAT --to-source %s"
+        % (dst_ip, dst_port, src_ip_port))
+
+
+def iptables_udp_divert_iface_traffic(iface, dst_port, target_dst_port):
+    """
+    iptables command to divert udp traffic egressing from an interface to a different port
+    Args:
+        iface: Interface packets are egressign from
+        dst_port: Destination port of packets to be diverted
+        target_dst_port: Destination port packets are diverted to
+    Returns: None
+    """
+    shellcmd = ShellCommandHelper()
+    shellcmd.run_cmd(
+        "sudo iptables -t nat -A OUTPUT -o %s -p udp --dport %s -j REDIRECT --to-ports %s"
+        % (iface, dst_port, target_dst_port))
+
+
+def create_vxlan_tunnel(vni, remote_ip, vxlan_port, vtep_ip):
+    """
+    Method to create a VxLAN tunnel and assign an IP to the VTEP
+    Args:
+        vni: VNI of tunnel to be created
+        remote_ip: Remote IP address for underlay network
+        vxlan_port: Remote destination port for VxLAN tunnel
+    Returns: None
+    """
+    shellcmd = ShellCommandHelper()
+    shellcmd.run_cmd(
+        "sudo ip link add vxlan type vxlan id %s remote %s dstport %s srcport %s %s nolearning"
+        % (vni, remote_ip, vxlan_port, vxlan_port, vxlan_port))
+    shellcmd.run_cmd("sudo ip addr add %s/24 dev vxlan" % (vtep_ip))
+    shellcmd.run_cmd("sudo ip link set vxlan up")
+
+
+def build_vxlan_ssh_conn(conn_params):
+    """
+    Args:
+        conn_params: JSON. Format:{
```

Comment:
Not important but I think this can be a dataclass instead.

Reply:
I was planning to turn it into a proto in the future. Will add a TODO.